### ISSUE TITLE
how to derive the expressions in ThreadoldL1 function when L1 regularization added ?

### ISSUE THREAD

**Issue by leoli2012** [dmlc/xgboost#1255]
hi, i read the source code,  found both  leaf weight calculate  function and cost of loss function added a ThresholdL1 method if alpha is not equal 0,  it should means cost function added L1 regularization, but i don't know how to  derived the calculation expression in ThreaholdL1 when L1 added. does anybody knows?

**Comment by pommedeterresautee** [comment]
Old question, closing
